src/core/geometry_utils.h

```diff
@@ -11,8 +11,35 @@
 #include <cmath>
 
 
-#define EQ_DOUBLE(a, b)                         \
-  (std::abs((a) - (b)) < 0.0001)
+inline bool equal(double a, double b) {
+  return (std::abs((a) - (b)) < 0.0001);
+}
+
+inline bool equal(double a, double b, double c) {
+  return (equal(a, b) && equal(a, c));
+}
+
+inline bool less_or_equal(double a, double b) {
+  return (equal(a, b) || (a < b));
+}
+
+inline bool is_between(double point, double left, double right) {
+  return (less_or_equal(left, point) && less_or_equal(point, right)) ||
+      (less_or_equal(right, point) && less_or_equal(point, left));
+}
+
+inline bool is_strictly_between(double point, double left, double right) {
```

Comment:
It's up to you. Both options are ok to me. In case you pick the second one use the "are_strictly_ordered" name.